packages/binarytree/src/proof.ts

```diff
@@ -0,0 +1,51 @@
+import { equalsBytes } from '@ethereumjs/util'
+
+import { createBinaryTree } from './constructors.js'
+import { decodeBinaryNode } from './node/index.js'
+
+import type { BinaryTree } from './binaryTree.js'
+import type { BinaryNode } from './node/index.js'
+import type { StemBinaryNode } from './node/stemNode.js'
+
+/**
+ * Saves the nodes from a proof into the tree.
+ * @param proof
+ */
+export async function binaryTreeFromProof(_proof: Uint8Array[]): Promise<BinaryTree> {
+  const proofTrie = await createBinaryTree()
+  const putStack: [Uint8Array, BinaryNode][] = _proof.map((bytes) => {
+    const node = decodeBinaryNode(bytes)
+    return [proofTrie['merkelize'](node), node]
+  })
+  await proofTrie.saveStack(putStack)
+  const root = putStack[0][0]
+  proofTrie.root(root)
+  return proofTrie
+}
+
+/**
+ * Verifies a proof.
+ * @param rootHash
+ * @param key
+ * @param proof
+ * @throws If proof is found to be invalid.
+ * @returns The value from the key, or null if valid proof of non-existence.
+ */
+export async function verifyBinaryProof(
+  _rootHash: Uint8Array,
```

Comment:
Why are there `_` before these variable names?  It looks like they are all used in the code so assuming we don't need this for linting, right?

Reply:
Probably put there to satisfy the linter when the method was not yet implemented.  can delete the `_` now